src/v/archival/service.cc

```diff
@@ -472,10 +474,30 @@ ss::future<> scheduler_service_impl::run_uploads() {
                     .num_failed = lhs.num_failed + rhs.num_failed};
               });
 
-            if (total.num_succeded == 0 && total.num_failed == 0) {
+            _probe.successful_upload(total.num_succeded);
+            _probe.failed_upload(total.num_failed);
+
+            if (total.num_failed != 0) {
+                vlog(
+                  archival_log.error,
+                  "Failed to upload {} segments out of {}",
+                  total.num_failed,
+                  total.num_succeded + total.num_failed);
+            } else if (total.num_failed != 0) {
```

Comment:
this doesn't make sense to me. i think is a bug

`total.num_succeeded` can it both fail and succeed and do you reset them? if so where.

Reply:
that's supposed to be `num_succeded`
the purpose of this check is to limit logging if nothing happened (0 failed and 0 uploaded)
fixed

Reply:
yeh this looks like a typo or bug